cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -159,13 +168,16 @@ void DrawableSubmap::SetAlpha(const double current_tracking_z) {
 
 void DrawableSubmap::UpdateSceneNode() {
   ::cartographer::common::MutexLocker locker(&mutex_);
+  render_in_progress_ = false;
+  if (quitting_) {
```

Comment:
One small thing though. Shouldn't reading of query_in_progress_ in the destructor be protected with a mutex? Imagine that destructor reads query_in_progress == false, and immediately after, MaybeFetchTexture() in another thread initializes it.